python_scripts/functions.py

```diff
@@ -269,11 +263,48 @@ def local_file_to_s3(local_path, s3_path):
     with open(local_path, "rb") as f:
         s3_client.upload_fileobj(f, b, o)
 
+
+
 def generate_iam_config(config, outpath="iam_config.yaml"):
     """
     Should take the necessary paths in the config and write out an
     iam that has access to those s3 paths e.g.
     write-only to log-base-path
     read/write to land-base-path
     """
-    pass
\ No newline at end of file
+
+    expected_keys = [
+        "iam-role-name",
+        "log-base-path",
+        "land-base-path",
+        "pass-base-path",
+        "fail-base-path"
+    ]
+
+    missing_keys = [key for key in expected_keys if key not in config]
+
+    if len(missing_keys) == 0:
```

Comment:
Should change to: `if not missing_keys:` according to PEP8. However if the config-schema checks this on read in you may not even need to check this.

Reply:
Nice PEP8 ref, good to know (for future historians, relevant section of PEP8: ```For sequences, (strings, lists, tuples), use the fact that empty sequences are false.```). As above, the config should already be validated so will remove this redundant check